### PR TITLE
disallow entry span to create propagation context

### PR DESCRIPTION
**Pull request by Shikugawa**
This PR is for entry span not to create propagation context packed to sw8 header. In general, we don't have to allow entry span to create them.

### CODE REVIEW

File: cpp2sky/segment_context.h [hunk 1]
```diff
@@ -238,15 +239,16 @@ class SegmentContext {
...
+  virtual std::optional<std::string> createSW8HeaderValue(
+      CurrentSegmentSpanPtr parent, std::string&& target_address) = 0;
   // If you don't specify parent span, stored to current segment, it will be
   // selected newest span as parent span.
-  virtual std::string createSW8HeaderValue(
+  virtual std::optional<std::string> createSW8HeaderValue(
       const std::string& target_address) = 0;
-  virtual std::string createSW8HeaderValue(std::string&& target_address) = 0;
+  virtual std::optional<std::string> createSW8HeaderValue(
+      std::string&& target_address) = 0;
```
Comment (wbpcode) [inline cpp2sky/segment_context.h:251]: nit: I'm not sure in what scenario `string&&` is needed as an argument type for the interface. Perhaps `const std::string_view` can be used to unify them.

**Review by wbpcode — COMMENTED** [review]
It's great. I think this is done well. Only one nit.

File: cpp2sky/segment_context.h (hunk 1 above)
Comment (Shikugawa) [reply cpp2sky/segment_context.h]: done

**Comment by wu-sheng** [comment]
Is this PR mistagged as 0.1.2? 0.1.2 has been released.

**Comment by wu-sheng** [comment]
> Disallow entry span to create propagation context to follow STAM style.

Could you explain a little about this `disable`? I am not totally following by reading the codes.

**Comment by Shikugawa** [comment]
@wu-sheng This PR is for entry span not to create propagation context packed to `sw8` header. In general, we don't have to allow entry span to create them.

**Comment by wu-sheng** [comment]
> @wu-sheng This PR is for entry span not to create propagation context packed to `sw8` header. In general, we don't have to allow entry span to create them.

OK, yes, we should not create that, context is only created as exit span.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**